### ISSUE TITLE
Docker?

### ISSUE THREAD

**Issue by SpokeyWheeler** [DMPRoadmap/roadmap#3282]
Please complete the following fields as applicable:

**What version of the DMPRoadmap code are you running? (e.g. v2.2.0)**

None yet

**Expected behaviour:**

A preconfigured docker image that I can pull to test the application

**Actual behaviour:**

No docker image

**Steps to reproduce:**

**Comment by briri** [comment]
Hi @SpokeyWheeler 

I agree that it would be a nice to have a docker image to help people get up and running quickly with the codebase. There was a [separate repository](https://github.com/DMPRoadmap/roadmap-devenv) that was created a few years ago to try and accomplish this, but it has since been archived.

Each fork of the codebase has their own approaches to deploying and supporting their production instances. They are usually very specific though to their situations. We also do not have dedicated full time resources and so will likely not get around to building a base Docker image in the near future. If you have time and would be will to set one up and contribute it back to this project we would greatly appreciate it!

**Comment by SpokeyWheeler** [comment]
I can have a stab at it, but it's more the configuration that needs to be done that I might struggle with.

**Comment by H-Kwon** [comment]
Hi @SpokeyWheeler,

Is there any chance that you made some progress on it? I am about to try to install DMPRoadmap on my local for the first time now. But I was wondering if there was some Dockerfile or image I could reuse in case I set up a new environment remotely.

**Comment by SpokeyWheeler** [comment]
Hi @H-Kwon 

Sorry, I can't get it built, hence why I wanted the Docker image.

**Comment by briri** [comment]
@SpokeyWheeler and @H-Kwon I created this repo last summer to play around with some of the AWS CloudFormation templating technology. I had to create a Dockerfile in order to get DMPRoadmap up and running in an elastic container service environment.

I'm not sure if it's still in a working state, but it should give you a good starting point for you. https://github.com/CDLUC3/dmp-roadmap-cfn/blob/main/application/Dockerfile

**Comment by SpokeyWheeler** [comment]
What does 
` COPY roadmap/ /roadmap`
do?

>  => ERROR [ 8/28] COPY roadmap/ /roadmap                                                                                                                                0.0s
------
 > [ 8/28] COPY roadmap/ /roadmap:
------
> failed to compute cache key: failed to calculate checksum of ref ik4hwmok88oo1wq3wgbhwxen4::e9ua61kdx1pll3eafp5jicd34: "/roadmap": not found

**Comment by H-Kwon** [comment]
@SpokeyWheeler You need to clone the entire repo of https://github.com/CDLUC3/dmp-roadmap-cfn.git.

```
$ git clone https://github.com/CDLUC3/dmp-roadmap-cfn.git
$ cd dmp-roadmap-cfn/
$ git submodule update --init --recursive
$ cd application
$ docker build . -t roadmap:685cd9
```

There are 2 things you need to keep in mind:

1. The roadmap/ directory points to 685cd9(committed Tue Jul 5 16:28:16 2022 -0700)
2. You still need to set up a database separately.

**Comment by briri** [comment]
Yes, thank you @H-Kwon. The reference I provided was meant to be used as an example not a solution you'd be able to run with. 
The latest version of the DMPRoadmap code has a new setup script you can use to seed the DB and setup the correct configuration for an initial install. to do that run `ruby bin/setup.rb postgres ` (or `mysql` if you prefer to use that DB). As @H-Kwon noted, you will need to have a DB running somewhere. Whether that is in the same Docker image or a different one is up to you.

**Comment by SpokeyWheeler** [comment]
Sorry, I did manage to get past that, but eventually I ran into an issue that required sudo, and my work laptop is locked down, so I gave up.
